packages/terra-paginator/src/_PaginatorButton.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import * as KeyCode from 'keycode-js';
+import styles from './Paginator.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  children: PropTypes.node,
+  /**
+   * Callback function triggered when clicked.
+   */
+  onClick: PropTypes.func,
+};
+
+class PaginatorButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { active: false, focused: false };
+    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+    this.handleOnKeyUp = this.handleOnKeyUp.bind(this);
+    this.handleOnBlur = this.handleOnBlur.bind(this);
+    this.handleFocus = this.handleFocus.bind(this);
+    this.handleMouseDown = this.handleMouseDown.bind(this);
+
+    this.shouldShowFocus = true;
+  }
+
+  handleOnBlur() {
+    this.setState({ focused: false });
+  }
+
+  handleOnKeyDown(event) {
+    // Add active state to FF browsers
+    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE) {
+      this.setState({ active: true });
+    }
+
+    // Add focus styles for keyboard navigation
+    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN) {
+      this.setState({ focused: true });
```

Comment:
Nice work @amichaelparker 
Let's update the approach from internal state to using a data attribute instead, similar to terra-application-navigation and terra-list:

Example here of it as a utility:
https://github.com/cerner/terra-framework/blob/c28c37b8a7030143cccfd4970d8d75b8885ed677/packages/terra-application-navigation/src/utils/helpers.js#L9-21

Example here of it in use (using onBlur and onMouseDown):
https://github.com/cerner/terra-framework/blob/c28c37b8a7030143cccfd4970d8d75b8885ed677/packages/terra-application-navigation/src/utility-menu/_UtilityMenuHeaderButton.jsx#L65-L66


And corresponding CSS to match:
https://github.com/cerner/terra-framework/blob/c28c37b8a7030143cccfd4970d8d75b8885ed677/packages/terra-application-navigation/src/utility-menu/UtilityMenuHeaderButton.module.scss#L29-34

Reply:
I moved everything over, and it now uses the data attributes to apply the styles. The button doesn't strictly need to be a separate private component at this point, but I think it also keeps some of the clutter out of the main pagination component.